### PR TITLE
Add survey task chooser, choices, and choice button

### LINKED ISSUE #1903: Add survey Choices component
Add survey Choices component, to display sorted/filtered/selected choices.

See https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/tasks/survey/chooser/Choices.jsx.

### PR DESCRIPTION
**Pull request by mcbouslog**
Package: `lib-classifier`

Closes #1903, towards #1901.

Related staging project: https://local.zooniverse.org:8080/?project=1634&workflow=2434

Describe your changes:
- [x] adds basic Chooser component (container for filters and choices)
- [x] adds Choices component
- [x] adds Choices storybooks
- [x] adds ChoiceButton
- [x] add (dynamically sized) ChoiceButton thumbnail
- [x] adds ChoiceButton stories

Open items:
- [x] tests

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js:45]: PFE uses [`flex-basis` applied to all children](https://github.com/zooniverse/Panoptes-Front-End/blob/master/css/survey-task.styl#L119-L127) of the containing element, depending on a column property. I originally implemented the same as PFE, and can easily revert accordingly, but on a second try using `<Grid />` it worked very cleanly, so running with `<Grid />` initially, but very open to suggestion.

Comment (eatyourgreens) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js:45]: Grids support [auto-placement by column](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Auto-placement_in_CSS_Grid_Layout#auto-placement_by_column), which would allow us to get rid of the old hack for splitting the choices list into smaller column lists. My initial vote would be for grids.

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js:57]: Marking the list of choices up as a HTML list might make it easier to navigate with a screen reader.

Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js]: Great suggestion, thank you @eatyourgreens! I hope it's ok I add this to #1991? I'm hopeful it'll be easier to test accessibility, focus, keys, etc. once all the survey task pieces are wired together, and it'll keep the initial PRs, like this one, just a little simpler.

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices/Choices.stories.js
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices/Choices.stories.js:6]: Not ideal file or folder naming and duplicates mock survey task with task model and annotation model. Major refactor (to utilize factory?) can come in subsequent PR, but maybe naming, folder, or data formatting cleanup should be included in this PR?

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices/Choices.stories.js
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices/Choices.stories.js:52]: Should add a story for more than 30 choices (no thumbnails), but the mock survey task currently using (from staging workflow, built from project builder example CSVs) has less than 30 choices, and getting to greater than 30 choices will take some work, so would like to postpone for now.

```suggestion
// add MoreThanThirty story with refactored mock survey task
```

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js
Comment (eatyourgreens) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js]: Sounds good to me. By the way, we have a volunteer who classifies on survey projects, with audio subjects, using NVDA.

Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js]: That's awesome! Do you know which project? I'll make a note to test on.

Comment (eatyourgreens) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js]: https://www.zooniverse.org/talk/18/1002723?comment=1884567

Comment (eatyourgreens) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js]: There was also a volunteer with a severe mobility impairment, who emailed us via `contact@`, who dictates his survey task classifications using Dragon. Continuing to support voice control would be good.

Comment (eatyourgreens) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js]: From memory, changes that would help in a screen reader are meaningful alt text for example images in the `Choice` component, or the ability for audio projects to embed audio files, instead of images, for each choice.

**Comment by mcbouslog** [comment]
@srallen this is ready for initial cursory review, which we discussed (this comment isn't necessary), but I thought it would be good for me to set a precedent for subsequent related survey task PRs (like #1989, which is not ready for initial cursory review just yet, but I'll tag and post a similar comment when it is)

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js
Comment (eatyourgreens) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices.js]: For a sighted user, we could maybe distinguish between radio buttons and checkboxes for the answers to Choice questions. At the moment they look exactly the same, which makes them hard to navigate with the keyboard.

**Review by srallen — DISMISSED** [review]
I think the basic setup is fine here. I have some non-blocking suggestions, but could be address in follow up PRs for specific components. I'm curious if there's a utility method in lodash that we can use in place of the custom `sort-into-columns` library we have. I don't CSS Grid will replace this (could be wrong) since what it is doing is sorting the items in the array from something like `a, b, c, d, e, f` to `a, d, b, e, c, f` assuming we're in the 2 column layout. I've taken screenshots just to note in this PR for future reference  what I mean.

Without `sort-into-columns`:
Screen Shot 2021-01-20 at 4 03 10 PM

With `sort-into-columns`:
Screen Shot 2021-01-20 at 4 03 26 PM

**Comment by eatyourgreens** [comment]
Grid auto columns should be able to handle the second layout. That could be worth investigating separately. https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Auto-placement_in_CSS_Grid_Layout#auto-placement_by_column

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices/components/ChoiceButton.js
Comment (srallen) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Chooser/components/Choices/components/ChoiceButton.js:18]: Addressing this can be done separately in a follow up PR.

**Comment by srallen** [comment]
I've taken this over to write tests.

**Comment by srallen** [comment]
Ok I've finished writing the tests. Enough has changed that I think this warrants another review from someone else. @mcbouslog could you dismiss my review as stale? @ErikOstlund do you have time this week to review this? Thanks!

**Comment by srallen** [comment]
I'll look into the CI failure today.

**Comment by ErikOstlund** [comment]
Testing this PR in a local environment at `https://local.zooniverse.org:8080/?project=1634&workflow=2434`
on Chrome

✅   Classifying the images work as expected.

❓ The Filter is not working for me. Is this suppose to be functional?
image

❓ Is there a way to break-up long words to prevent odd break points?
image


✅. Storybook components look good.

✅. All unit tests are passing.

**Comment by srallen** [comment]
@ErikOstlund 

> The Filter is not working for me. Is this suppose to be functional?

@mcbouslog is working on functionality in #1989, so this is visual placeholder right now.

> ❓ Is there a way to break-up long words to prevent odd break points?

It's styled with `word-break: break-word;` right now. The PFE version just lets buttons overlap. Check out how `
Hippopotamus` displays right now there:

https://master.pfe-preview.zooniverse.org/projects/markb-panoptes/survey-task-test-project/classify

Neither are a good solution. I've advised @mcbouslog to do a design pass asking Becky for suggestions for stuff like this in a follow up PR. Sound good?

**Review by ErikOstlund — APPROVED** [review]
The Filter and word-break issues are being addressed in separate PRs.
Everything else looks good!

**Pull request merged**
